apis/Google.Cloud.Firestore/Google.Cloud.Firestore/Query.cs

```diff
@@ -509,6 +511,50 @@ object ConvertValueToDocumentReferencesForInQuery()
             }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        public Query Where(Filter filter)
+        {
+            var newFilters = _filters == null ? new List<InternalFilter>() : new List<InternalFilter>(_filters);
+            if(filter.GetType() == typeof(Filter.UnaryFilter))
+            {
+                newFilters.Add(ParseFieldFilter((Filter.UnaryFilter) filter));
+            }
+            else
+            {
+                newFilters.Add(ParseCompositeFilter((Filter.CompositeFilter) filter));
+            }
+            
+            return new Query(_root, _offset, _limit, _orderings, newFilters, _projections, _startAt, _endAt);
+        }
+
+        private InternalFilter ParseFieldFilter(Filter.UnaryFilter unaryFilter)
```

Comment:
This isn't really parsing anything in the normal sense. It may not be needed in end anyway, if everything is rewritten in terms of Filter.